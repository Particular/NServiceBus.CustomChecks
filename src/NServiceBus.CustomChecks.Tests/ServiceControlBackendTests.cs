namespace NServiceBus.CustomChecks.Tests
{
    using System;
    using System.Text;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using CustomChecks;
    using NUnit.Framework;
    using ServiceControl.Plugin.CustomChecks.Messages;

    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class ServiceControlBackendTests
    {
        [Test]
        public void It_can_serialize_RegisterCustomCheckResult()
        {
            var body = ServiceControlBackend.Serialize(new ReportCustomCheckResult
            {
                EndpointName = "My.Endpoint",
                ReportedAt = new DateTime(2016, 02, 01, 13, 59, 0, DateTimeKind.Utc),
                Host = "Host",
                HostId = Guid.Empty,
                Category = "MyCategory",
                CustomCheckId = "MyCheckId",
                FailureReason = "Failure reason.",
                HasFailed = true
            });
            Approvals.Verify(Encoding.UTF8.GetString(body));
        }
    }
}