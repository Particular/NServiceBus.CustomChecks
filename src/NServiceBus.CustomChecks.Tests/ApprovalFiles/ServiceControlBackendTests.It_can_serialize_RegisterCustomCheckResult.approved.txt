{"HostId":"00000000-0000-0000-0000-000000000000","CustomCheckId":"MyCheckId","Category":"MyCategory","HasFailed":true,"FailureReason":"Failure reason.","ReportedAt":"2016-02-01T13:59:00Z","EndpointName":"My.Endpoint","Host":"Host"}